File: test-suite/src/test/java/io/micronaut/docs/qualifiers/replaces/defaultimpl/DefaultImplementationSpec.java
package io.micronaut.docs.qualifiers.replaces.defaultimpl;

import io.micronaut.context.BeanContext;
import io.micronaut.context.DefaultBeanContext;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class DefaultImplementationSpec {

    @Test
    void testTheDefaultIsReplaced() {
        BeanContext ctx = BeanContext.run();
        Assertions.assertTrue(ctx.getBean(ResponseStrategy.class) instanceof CustomResponseStrategy);
        ctx.close();
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/qualifiers/replaces/defaultimpl/DefaultImplementationSpec.kt
package io.micronaut.docs.qualifiers.replaces.defaultimpl

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.types.shouldBeInstanceOf
import io.micronaut.context.BeanContext

class DefaultImplementationSpec : StringSpec({

    "test the default implementation is replaced" {
        val ctx = BeanContext.run()
        val responseStrategy = ctx.getBean(ResponseStrategy::class.java)

        responseStrategy.shouldBeInstanceOf<CustomResponseStrategy>()

        ctx.close()
    }
})
